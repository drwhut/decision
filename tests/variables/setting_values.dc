Start~#1

[Variable(myIntegerVar, Integer, 69)]
[Variable(myFloatVar, Float, 420.69)]
[Variable(myStringVar, String, "Hello, world!")]
[Variable(myBooleanVar, Boolean, true)]

> setting and printing an integer variable
myIntegerVar~#2
Add(#2, 1)~#3
Set(myIntegerVar, #1, #3)~#4
Print(#4, #2)~#5

> setting and printing a float variable
myFloatVar~#6
Divide(#6, 2)~#7
Set(myFloatVar, #5, #7)~#8
Print(#8, #6)~#9

> setting and printing a string variable
> TODO: Implement a node to manipulate strings
myStringVar~#10
Set(myStringVar, #9, "Goodbye, world!")~#11
Print(#11, #10)~#12

> setting and printing a boolean variable
myBooleanVar~#13
Xor(#13, true)~#14
Set(myBooleanVar, #12, #14)~#15
Print(#15, #13)~#16