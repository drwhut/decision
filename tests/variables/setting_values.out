70
210.345
Goodbye, world!
false
